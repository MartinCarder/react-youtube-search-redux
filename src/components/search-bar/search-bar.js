import React from 'react'
import { SearchBarWrapper, StyledForm, SearchInput, SearchButton } from './search-bar.styles'

const SearchBar = () => {
  let formRef = null

  const submitSearch = ev => {
    ev.preventDefault()
    console.log(formRef.elements.search.value)
  }

  return (
    <SearchBarWrapper>
      <StyledForm innerRef={ref => (formRef = ref)}>
        <SearchInput name="search" type="text" placeholder="Search youtube" />
        <SearchButton onClick={submitSearch}>Search</SearchButton>
      </StyledForm>
    </SearchBarWrapper>
  )
}

export default SearchBar
