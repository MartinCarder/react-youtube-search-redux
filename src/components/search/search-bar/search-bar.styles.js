import styled, { css } from 'styled-components'

export const SearchBarWrapper = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  border-bottom: 1px solid #d5d5d5;
  background-color: #ffffff;
  z-index: 2;
  padding: 10px 5px;
`

export const StyledForm = styled.form`
  max-width: 500px;
  margin: 0 auto;
  display: flex;
`

const searchItem = css`
  height: 30px;
  box-sizing: border-box;
`

export const SearchInput = styled.input`
  ${searchItem};
  flex: 1 1 0;
  padding: 0 5px;
  font-size: 14px;
`

export const SearchButton = styled.button`
  ${searchItem};
  flex: 0 0 70px;
`
