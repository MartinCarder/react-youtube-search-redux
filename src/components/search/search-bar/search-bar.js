import React, { PureComponent } from 'react'
import { SearchBarWrapper, StyledForm, SearchInput, SearchButton } from './search-bar.styles'

class SearchBar extends PureComponent {
  state = {
    search: '',
  }

  submitSearch = ev => {
    ev.preventDefault()
    const { search } = this.state
    const { startSearch } = this.props

    startSearch(search)
  }

  handleSearchChange = event => {
    this.setState({ search: event.target.value })
  }

  render() {
    return (
      <SearchBarWrapper>
        <StyledForm>
          <SearchInput
            name="search"
            type="text"
            placeholder="Search youtube"
            value={this.state.name}
            onChange={this.handleSearchChange}
          />
          <SearchButton onClick={this.submitSearch}>Search</SearchButton>
        </StyledForm>
      </SearchBarWrapper>
    )
  }
}

export default SearchBar
