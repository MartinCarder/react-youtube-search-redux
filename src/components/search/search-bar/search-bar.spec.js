import React from 'react'
import { shallow, render } from 'enzyme'
import SearchBar from './search-bar'
import { SearchInput, SearchButton } from './search-bar.styles'

describe('<SearchBar>', () => {
  it('Component renders correctly', () => {
    const wrapper = render(<SearchBar />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Returns value in text input on search', () => {
    const startSearch = jest.fn()
    const wrapper = shallow(<SearchBar startSearch={startSearch} />)
    const CATS = 'cats'

    wrapper
      .find(SearchInput)
      .first()
      .simulate('change', { target: { value: CATS } })
    wrapper
      .find(SearchButton)
      .first()
      .simulate('click', { preventDefault: () => {} })

    expect(startSearch.mock.calls[0][0]).toBe(CATS)
  })
})
