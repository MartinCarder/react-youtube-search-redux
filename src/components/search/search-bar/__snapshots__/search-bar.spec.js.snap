// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar> Component renders correctly 1`] = `
.c0 {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  border-bottom: 1px solid #d5d5d5;
  background-color: #ffffff;
  z-index: 2;
  padding: 10px 5px;
}

.c1 {
  max-width: 500px;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  height: 30px;
  box-sizing: border-box;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  padding: 0 5px;
  font-size: 14px;
}

.c3 {
  height: 30px;
  box-sizing: border-box;
  -webkit-flex: 0 0 70px;
  -ms-flex: 0 0 70px;
  flex: 0 0 70px;
}

<div
  class="c0"
>
  <form
    class="c1"
  >
    <input
      class="c2"
      name="search"
      placeholder="Search youtube"
      type="text"
    >
      
    </input>
    <button
      class="c3"
    >
      Search
    </button>
  </form>
</div>
`;
